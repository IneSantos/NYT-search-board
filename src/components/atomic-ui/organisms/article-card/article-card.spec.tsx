import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import ArticleCard from "./article-card";

let store;
let mockedStore;

describe("ArticleCard component tests", () => {
  test(`ArticleCard component render`, () => {
    const article = {
      id: "",
      headline: "This is a headline",
      abstract: "",
      keywords: [],
      section: "",
      snippet: "",
      pub_date: "",
      source: "",
      web_url: "",
    };
    mockedStore = configureStore();
    store = mockedStore({
      article: {
        articleList: [article],
        page: 0,
        searchClicked: false,
        article: article,
      },
    });

    const { container } = render(
      <Router>
        <Provider store={store}>
          <ArticleCard article={article} index={1} />
        </Provider>
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
