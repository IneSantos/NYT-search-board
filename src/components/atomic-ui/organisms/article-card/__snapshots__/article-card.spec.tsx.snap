// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard component tests ArticleCard component render 1`] = `
<a
  href="/article/1"
  style="text-decoration: none;"
>
  <div
    class="card-wrapper"
  >
    <div
      class="card-header"
    >
      <h3
        class="title-small"
      >
        This is a headline
      </h3>
    </div>
  </div>
</a>
`;
